commit d7dcb9a104b99cc98bba9aaf4f3bac4031ce8a8b (refs/changes/31/5631/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2019-02-22T23:27:53+00:00 (8 months ago)
    
    joyent/bugview#31 Support numbered lists
    Reviewed by: Mike Gerdts <mike.gerdts@joyent.com>
    Approved by: Mike Gerdts <mike.gerdts@joyent.com>

diff --git a/jirapub.js b/jirapub.js
index ad40cc3..72ddf4e 100644
--- a/jirapub.js
+++ b/jirapub.js
@@ -656,15 +656,28 @@ parse_jira_markup(desc, ps)
 				leading_spaces++;
 				continue;
 			} else if ((c === '*' || c === '-') && cc === ' ') {
-				if (!ps.ps_list) {
+				if (ps.ps_list) {
+					out.push('</li>');
+				} else {
 					out.push('<ul>');
 				}
-				ps.ps_list = true;
+				ps.ps_list = 'ul';
+				commit_text();
+				out.push('<li>');
+				continue;
+			} else if (c === '#' && cc === ' ') {
+				if (ps.ps_list) {
+					out.push('</li>');
+				} else {
+					out.push('<ol>');
+				}
+				ps.ps_list = 'ol';
 				commit_text();
 				out.push('<li>');
 				continue;
 			}
 
+
 			/*
 			 * No special sequence was detected, so emit the
 			 * spaces we counted, switch to the TEXT state, and
@@ -679,8 +692,8 @@ parse_jira_markup(desc, ps)
 		case 'TEXT':
 			if (ps.ps_list && i === 0 && c !== ' ') {
 				commit_text();
-				ps.ps_list = false;
-				out.push('</ul>');
+				out.push('</li></' + ps.ps_list + '>');
+				ps.ps_list = null;
 
 				/*
 				 * Note that we must break out here, so that
